### ISSUE TITLE
[JENKINS-28155] Job fails with [An existing connection was forcibly closed by the remote host]

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#9736]
<p>Our Windows slaves are JNLP connected using scheduled tasks (like this: <a href="https://wiki.jenkins-ci.org/display/JENKINS/Launch+Java+Web+Start+slave+agent+via+Windows+Scheduler" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Launch+Java+Web+Start+slave+agent+via+Windows+Scheduler</a>).</p>

<p>When a job starts working with a slave it suddenly fails because the connection was lost. It says something on slave side killed the connection. Few minutes afterwards the slave is back on-line. I am sure it is not a network issue as I tried pinging the network from system start-up till the problem occurs. I could not find find what kills the connection and disable TCP chimney (advice from google) did not work.</p>

<p>The exception is below.</p>

<p>I think proper behaviour would be not to fail the job but attempt to reconnect. Any hints of additional configuration that will help me resolve this issue is appreciated.</p>

<p>Building remotely on vmbam32.eur.ad.sag (R2 Server 2012 Enterprise Windows) in workspace c:\jenkins\workspace\optimize-install<br/>
FATAL: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@​b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span><br/>
hudson.remoting.RequestAbortedException: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@​b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span><br/>
	at hudson.remoting.Request.abort(Request.java:296)<br/>
	at hudson.remoting.Channel.terminate(Channel.java:815)<br/>
	at hudson.remoting.Channel$2.terminate(Channel.java:492)<br/>
	at hudson.remoting.AbstractByteArrayCommandTransport$1.terminate(AbstractByteArrayCommandTransport.java:72)<br/>
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)<br/>
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)<br/>
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)<br/>
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)<br/>
	at java.util.concurrent.FutureTask.run(Unknown Source)<br/>
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)<br/>
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)<br/>
	at java.lang.Thread.run(Unknown Source)<br/>
	at ......remote call to vmbam32.eur.ad.sag(Native Method)<br/>
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1361)<br/>
	at hudson.remoting.Request.call(Request.java:171)<br/>
	at hudson.remoting.Channel.call(Channel.java:752)<br/>
	at hudson.FilePath.act(FilePath.java:980)<br/>
	at hudson.FilePath.act(FilePath.java:969)<br/>
	at hudson.FilePath.mkdirs(FilePath.java:1152)<br/>
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1269)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:610)<br/>
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)<br/>
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:532)<br/>
	at hudson.model.Run.execute(Run.java:1744)<br/>
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:98)<br/>
	at hudson.model.Executor.run(Executor.java:374)<br/>
Caused by: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@​b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span><br/>
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)<br/>
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)<br/>
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)<br/>
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)<br/>
	at java.util.concurrent.FutureTask.run(Unknown Source)<br/>
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)<br/>
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)<br/>
	at java.lang.Thread.run(Unknown Source)<br/>
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host<br/>
	at sun.nio.ch.SocketDispatcher.read0(Native Method)<br/>
	at sun.nio.ch.SocketDispatcher.read(Unknown Source)<br/>
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(Unknown Source)<br/>
	at sun.nio.ch.IOUtil.read(Unknown Source)<br/>
	at sun.nio.ch.SocketChannelImpl.read(Unknown Source)<br/>
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)<br/>
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)<br/>
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:561)<br/>
	... 6 more</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vassilena">vassilena</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-28155" target="_blank">Job fails with [An existing connection was forcibly closed by the remote host]</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core, remoting, windows-slaves-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2022-11-22T23:13:03+00:00
<li><b>votes</b>: 27
<li><b>watchers</b>: 30
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Our Windows slaves are JNLP connected using scheduled tasks (like this: <a href="https://wiki.jenkins-ci.org/display/JENKINS/Launch+Java+Web+Start+slave+agent+via+Windows+Scheduler" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Launch+Java+Web+Start+slave+agent+via+Windows+Scheduler</a>).</p>

<p>When a job starts working with a slave it suddenly fails because the connection was lost. It says something on slave side killed the connection. Few minutes afterwards the slave is back on-line. I am sure it is not a network issue as I tried pinging the network from system start-up till the problem occurs. I could not find find what kills the connection and disable TCP chimney (advice from google) did not work.</p>

<p>The exception is below.</p>

<p>I think proper behaviour would be not to fail the job but attempt to reconnect. Any hints of additional configuration that will help me resolve this issue is appreciated.</p>

<p>Building remotely on vmbam32.eur.ad.sag (R2 Server 2012 Enterprise Windows) in workspace c:\jenkins\workspace\optimize-install
FATAL: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span>
hudson.remoting.RequestAbortedException: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span>
	at hudson.remoting.Request.abort(Request.java:296)
	at hudson.remoting.Channel.terminate(Channel.java:815)
	at hudson.remoting.Channel$2.terminate(Channel.java:492)
	at hudson.remoting.AbstractByteArrayCommandTransport$1.terminate(AbstractByteArrayCommandTransport.java:72)
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
	at ......remote call to vmbam32.eur.ad.sag(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1361)
	at hudson.remoting.Request.call(Request.java:171)
	at hudson.remoting.Channel.call(Channel.java:752)
	at hudson.FilePath.act(FilePath.java:980)
	at hudson.FilePath.act(FilePath.java:969)
	at hudson.FilePath.mkdirs(FilePath.java:1152)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1269)
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:610)
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:532)
	at hudson.model.Run.execute(Run.java:1744)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@b7add1<span class="error">&#91;name=vmbam32.eur.ad.sag&#93;</span>
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(Unknown Source)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(Unknown Source)
	at sun.nio.ch.IOUtil.read(Unknown Source)
	at sun.nio.ch.SocketChannelImpl.read(Unknown Source)
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:561)
	... 6 more</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Slave Windows Server 2012 R<br/>
Master Windows Server 2012 R<br/>
Jenkins version 1.611
```
</details>
<details><summary><i>2 attachments</i></summary>

- [Jenkins Slave.xml](https://issues.jenkins.io/secure/attachment/30219/Jenkins%20Slave.xml)
- [sql-server-manager.jpg](https://issues.jenkins.io/secure/attachment/29733/sql-server-manager.jpg)
  > ![sql-server-manager.jpg](https://issues.jenkins.io/secure/attachment/29733/sql-server-manager.jpg)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jan_mueller_work.jpg" title="jan_mueller_work's avatar" /> jan_mueller_work's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=231329&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-231329">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>We may have a similar problem. We start a windows vm and connect it via jnlp to the master, then it starts some work and after some time, usually 3-5h the slave gets disconnected unintentionally:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@7aa84678[name=QF-WinXP]
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:211)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:631)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:564)
</pre>
</div></div>

<p>We have only one windows machine and this machine shows this behavior. All the linux slaves are connected constantly without any problems.</p>
</pre>
</details>

</details>
<p>We may have a similar problem. We start a windows vm and connect it via jnlp to the master, then it starts some work and after some time, usually 3-5h the slave gets disconnected unintentionally:</p>

<pre>
java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@​7aa84678[name=QF-WinXP]
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:211)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:631)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:745)
Caused by: java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:564)
</pre>

<p>We have only one windows machine and this machine shows this behavior. All the linux slaves are connected constantly without any problems.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/weakcamel.png" title="weakcamel's avatar" /> weakcamel's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=232682&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-232682">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><b>UPDATE</b>: It turned out my issue was related to the recent switch to https (so probably not related to this one).
After providing Java trustStore with RootCA and Intermediate CA - issue resolved.</p>

<p>====</p>

<p>Same here - although reason might be different. Slave started on WIndows 7, on an AD account, as a service.
Drops almost immediately.</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@240c0b4e[name=ST_windows_slave]
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:724)
Caused by: java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:561)
	... 6 more
</pre>
</div></div>



<p>Found some more in slave's log. My master instance is using https and although the certificate
is installed in Windows and browsers recognize it, slave complains about it - as below.
Any ideas would be welcome.</p>


<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Exception in thread "main" java.io.IOException: Failed to validate a server certificate. If you are using a self-signed certificate, you can use the -noCertificateCheck option to bypass this check.
	at hudson.remoting.Launcher.parseJnlpArguments(Launcher.java:327)
	at hudson.remoting.Launcher.run(Launcher.java:219)
	at hudson.remoting.Launcher.main(Launcher.java:192)
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.ssl.Alerts.getSSLException(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.fatal(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.ClientHandshaker.serverCertificate(Unknown Source)
	at sun.security.ssl.ClientHandshaker.processMessage(Unknown Source)
	at sun.security.ssl.Handshaker.processLoop(Unknown Source)
	at sun.security.ssl.Handshaker.process_record(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at hudson.remoting.Launcher.parseJnlpArguments(Launcher.java:269)
	... 2 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.validator.PKIXValidator.doBuild(Unknown Source)
	at sun.security.validator.PKIXValidator.engineValidate(Unknown Source)
	at sun.security.validator.Validator.validate(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.validate(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	... 14 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.provider.certpath.SunCertPathBuilder.build(Unknown Source)
	at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(Unknown Source)
	at java.security.cert.CertPathBuilder.build(Unknown Source)
	... 20 more
</pre>
</div></div>
</pre>
</details>

</details>
<p><b>UPDATE</b>: It turned out my issue was related to the recent switch to https (so probably not related to this one).<br/>
After providing Java trustStore with RootCA and Intermediate CA - issue resolved.</p>

<p>====</p>

<p>Same here - although reason might be different. Slave started on WIndows 7, on an AD account, as a service.<br/>
Drops almost immediately.</p>



```
java.io.IOException: Connection aborted: org.jenkinsci.remoting.nio.NioChannelHub$MonoNioTransport@​240c0b4e[name=ST_windows_slave]
	at org.jenkinsci.remoting.nio.NioChannelHub$NioTransport.abort(NioChannelHub.java:208)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:628)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:724)
Caused by: java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at org.jenkinsci.remoting.nio.FifoBuffer$Pointer.receive(FifoBuffer.java:136)
	at org.jenkinsci.remoting.nio.FifoBuffer.receive(FifoBuffer.java:306)
	at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:561)
	... 6 more
```



<p>Found some more in slave's log. My master instance is using https and although the certificate<br/>
is installed in Windows and browsers recognize it, slave complains about it - as below.<br/>
Any ideas would be welcome.</p>




```
Exception in thread "main" java.io.IOException: Failed to validate a server certificate. If you are using a self-signed certificate, you can use the -noCertificateCheck option to bypass this check.
	at hudson.remoting.Launcher.parseJnlpArguments(Launcher.java:327)
	at hudson.remoting.Launcher.run(Launcher.java:219)
	at hudson.remoting.Launcher.main(Launcher.java:192)
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.ssl.Alerts.getSSLException(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.fatal(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.Handshaker.fatalSE(Unknown Source)
	at sun.security.ssl.ClientHandshaker.serverCertificate(Unknown Source)
	at sun.security.ssl.ClientHandshaker.processMessage(Unknown Source)
	at sun.security.ssl.Handshaker.processLoop(Unknown Source)
	at sun.security.ssl.Handshaker.process_record(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at hudson.remoting.Launcher.parseJnlpArguments(Launcher.java:269)
	... 2 more
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.validator.PKIXValidator.doBuild(Unknown Source)
	at sun.security.validator.PKIXValidator.engineValidate(Unknown Source)
	at sun.security.validator.Validator.validate(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.validate(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	... 14 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at sun.security.provider.certpath.SunCertPathBuilder.build(Unknown Source)
	at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(Unknown Source)
	at java.security.cert.CertPathBuilder.build(Unknown Source)
	... 20 more
```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vassilena's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=232840&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-232840">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have no actual solution for this problem but after doing many different experiments we managed to configure our windows slaves (win server 2012 R2 standart) with a scheduled task (Jenkins Slave.xml attached here) and so far they are quite stable. A prerequisite is to have the test user automatically logged at OS boot.</p>
</pre>
</details>

</details>
<p>I have no actual solution for this problem but after doing many different experiments we managed to configure our windows slaves (win server 2012 R2 standart) with a scheduled task (Jenkins Slave.xml attached here) and so far they are quite stable. A prerequisite is to have the test user automatically logged at OS boot.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>franky4ro's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=235344&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-235344">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have started to have the same issues with the same configuration. I am using the LTS version of Jenkins 1.609.1</p>
</pre>
</details>

</details>
<p>I have started to have the same issues with the same configuration. I am using the LTS version of Jenkins 1.609.1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tjackson0's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=248820&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-248820">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Still an issue in Jenkins ver. 1.641</p>
</pre>
</details>

</details>
<p>Still an issue in Jenkins ver. 1.641</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/georgefischhof.jpg" title="georgefischhof's avatar" /> georgefischhof's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=263816&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-263816">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have the same issue: 
used: jenkins 2.09 (bitbnami stack)
master: win 7 64 bit, java 8.91 32 bit
slave: win 2008R2 64 bit, java 8.91 32 bit, 64 bit</p>
</pre>
</details>

</details>
<p>I have the same issue: <br/>
used: jenkins 2.09 (bitbnami stack)<br/>
master: win 7 64 bit, java 8.91 32 bit<br/>
slave: win 2008R2 64 bit, java 8.91 32 bit, 64 bit</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rbonaudo.png" title="rbonaudo's avatar" /> rbonaudo's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=284918&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-284918">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I had the same issue, also using the Windows' task scheduler to attach the slave.</p>

<p>For me, the issue seems to be due to the "Stop task if it runs longer than x time" option being activated and set to 3 days by default.
Windows kills the task after 3 days, stopping the connection between Jenkins and the slave.
This option is set in the properties of the task, inside Windows' task scheduler.</p>

<p>Hope this can help someone, I was blocked by this problem for a very long time myself.</p>
</pre>
</details>

</details>
<p>I had the same issue, also using the Windows' task scheduler to attach the slave.</p>

<p>For me, the issue seems to be due to the "Stop task if it runs longer than x time" option being activated and set to 3 days by default.<br/>
Windows kills the task after 3 days, stopping the connection between Jenkins and the slave.<br/>
This option is set in the properties of the task, inside Windows' task scheduler.</p>

<p>Hope this can help someone, I was blocked by this problem for a very long time myself.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>almahajan's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=286585&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286585">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hello,
We are also getting this exception with Jenkins ver. 2.38 inconsistently, has anyone found any working solution for this issue? Found <a href="https://issues.jenkins-ci.org/browse/JENKINS-27251" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-27251</a> where i see a comment saying "It was likely SocketTimeoutException on the slave side in this case (fixed in 2.60 &amp; released in 2.9)." We see "SocketTimeoutException" on slave side. So, does upgrading to 2.9 will solve this issue? </p>

<p>Thanks in advance!</p>
</pre>
</details>

</details>
<p>Hello,<br/>
We are also getting this exception with Jenkins ver. 2.38 inconsistently, has anyone found any working solution for this issue? Found <a href="https://issues.jenkins-ci.org/browse/JENKINS-27251" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-27251</a> where i see a comment saying "It was likely SocketTimeoutException on the slave side in this case (fixed in 2.60 & released in 2.9)." We see "SocketTimeoutException" on slave side. So, does upgrading to 2.9 will solve this issue? </p>

<p>Thanks in advance!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/georgefischhof.jpg" title="georgefischhof's avatar" /> georgefischhof's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=287569&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-287569">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi,</p>

<p>my working solution is the following:</p>

<p>I use windows for master and for the slave as well.</p>

<p>On the slave I have an administrator user with autologin, when the user logged in it waits 1 minute, then starts jenkins slave with java webstart with the jnlp file:
javaws.lnk c:\CI\slave-agent.jnlp 
(javaws.lnk is created to be on path)</p>

<p>I use java 1.7.1 because the javaws needs 1.7, but 1.7.89 has a security enhancement, and it does not allow to start the javaws automatically, it needs manual confirmation. (I did not check the versions between 1.7.1 and 1.7.89) 1.7.1 allows the starting.</p>

<p>This way the connection is stable.</p>

<p>BR,
George</p>
</pre>
</details>

</details>
<p>Hi,</p>

<p>my working solution is the following:</p>

<p>I use windows for master and for the slave as well.</p>

<p>On the slave I have an administrator user with autologin, when the user logged in it waits 1 minute, then starts jenkins slave with java webstart with the jnlp file:<br/>
javaws.lnk c:\CI\slave-agent.jnlp <br/>
(javaws.lnk is created to be on path)</p>

<p>I use java 1.7.1 because the javaws needs 1.7, but 1.7.89 has a security enhancement, and it does not allow to start the javaws automatically, it needs manual confirmation. (I did not check the versions between 1.7.1 and 1.7.89) 1.7.1 allows the starting.</p>

<p>This way the connection is stable.</p>

<p>BR,<br/>
George</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjdarton's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=308500&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-308500">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>We're also seeing much the same thing - the Jenkins server is saying that the (JNLP) slave dropped the connection.</p>

<p>We've been looking into this and we've found that (at least in our system) the problem is coming from the Slave's Windows OS itself.  On the slave, we are seeing the following error logged:</p>

<p><tt>java.io.IOException: An established connection was aborted by the software in your host machine.</tt>
 <tt>    at sun.nio.ch.SocketDispatcher.read0(Native Method)</tt>
 <tt>    at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:55)</tt>
 <tt>    at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:235)</tt>
 <tt>    at sun.nio.ch.IOUtil.read(IOUtil.java:209)</tt>
 <tt>    at sun.nio.ch.SockerChannelImpl.read(SocketChannelImpl.java:409)</tt>
 <tt>    at hudson.remoting.SocketChannelStream$1.read(SocketChannelStream.java:35)</tt>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:77)</tt>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:121)</tt>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:115)</tt>
 <tt>    at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)</tt>
 <tt>    at java.io.BufferedInputStream.read(BufferedInputStream.java:265)</tt>
 <tt>    at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:86)</tt>
 <tt>    at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:72)</tt>
 <tt>    at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:103)</tt>
 <tt>    at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:39)</tt>
 <tt>    at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)</tt>
 <tt>    at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:59)</tt></p>

<p>Some searching later, we identified that "An established connection was aborted by the software in your host machine" is the Windows socket error WSACONNABORTED (10053).</p>

<p>Much investigation later, we've found that various Windows services running on the Slave's OS are logging (in the Windows event log) that they're restarting at the exact same time, and (more interestingly!) we've also seen that the DHCP lease was renewed at the exact time that the slave reported the connection had died.</p>
<ul>
	<li>Note: <b>Windows handles TCP connections differently to other operating systems</b>.  If Windows decides that the physical network layer has gone down (however briefly), it actively kills (with a WSACONNABORTED) all TCP connections that were being routed over that network, thus turning a transient outage (that normal TCP retransmissions would handle so that the user doesn't even see the problem) into an application-level outage (as the TCP connection closes, forcing the application to deal with it, usually by reporting that the connection has failed and it's "game over").  This is why a brief network outage that should cause no operational impact will result in a flurry of service restarts as they all try to handle the connection losses.  Windows' (mis?)handling of this scenario has been like this for so long that I doubt Microsoft would be willing to change it now.</li>
</ul>


<p>So I think that something, somewhere deep within Windows, is making Windows believe that it has lost the network layer.  Problem is, I don't (yet) know what's doing it - all I see is a lot of symptoms of it doing it, not the root cause.</p>

<p>"ipconfig /release &amp;&amp; ipconfig /renew" will cause this (even if you immediately get the same IP address back), as will unplugging/replugging your Cat5 cable or disconnecting/reconnecting your WiFi connection, or power-saving on your NIC, or reconfiguring your NIC and thus forcing a reload of the driver, or...</p>

<p>We've yet to find the root cause in our setup, but investigations are ongoing.</p>
</pre>
</details>

</details>
<p>We're also seeing much the same thing - the Jenkins server is saying that the (JNLP) slave dropped the connection.</p>

<p>We've been looking into this and we've found that (at least in our system) the problem is coming from the Slave's Windows OS itself.  On the slave, we are seeing the following error logged:</p>

<p><tt>java.io.IOException: An established connection was aborted by the software in your host machine.</tt><br/>
 <tt>    at sun.nio.ch.SocketDispatcher.read0(Native Method)</tt><br/>
 <tt>    at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:55)</tt><br/>
 <tt>    at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:235)</tt><br/>
 <tt>    at sun.nio.ch.IOUtil.read(IOUtil.java:209)</tt><br/>
 <tt>    at sun.nio.ch.SockerChannelImpl.read(SocketChannelImpl.java:409)</tt><br/>
 <tt>    at hudson.remoting.SocketChannelStream$1.read(SocketChannelStream.java:35)</tt><br/>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:77)</tt><br/>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:121)</tt><br/>
 <tt>    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:115)</tt><br/>
 <tt>    at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)</tt><br/>
 <tt>    at java.io.BufferedInputStream.read(BufferedInputStream.java:265)</tt><br/>
 <tt>    at hudson.remoting.FlightRecorderInputStream.read(FlightRecorderInputStream.java:86)</tt><br/>
 <tt>    at hudson.remoting.ChunkedInputStream.readHeader(ChunkedInputStream.java:72)</tt><br/>
 <tt>    at hudson.remoting.ChunkedInputStream.readUntilBreak(ChunkedInputStream.java:103)</tt><br/>
 <tt>    at hudson.remoting.ChunkedCommandTransport.readBlock(ChunkedCommandTransport.java:39)</tt><br/>
 <tt>    at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)</tt><br/>
 <tt>    at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:59)</tt></p>

<p>Some searching later, we identified that "An established connection was aborted by the software in your host machine" is the Windows socket error WSACONNABORTED (10053).</p>

<p>Much investigation later, we've found that various Windows services running on the Slave's OS are logging (in the Windows event log) that they're restarting at the exact same time, and (more interestingly!) we've also seen that the DHCP lease was renewed at the exact time that the slave reported the connection had died.</p>
<ul>
	<li>Note: <b>Windows handles TCP connections differently to other operating systems</b>.  If Windows decides that the physical network layer has gone down (however briefly), it actively kills (with a WSACONNABORTED) all TCP connections that were being routed over that network, thus turning a transient outage (that normal TCP retransmissions would handle so that the user doesn't even see the problem) into an application-level outage (as the TCP connection closes, forcing the application to deal with it, usually by reporting that the connection has failed and it's "game over").  This is why a brief network outage that should cause no operational impact will result in a flurry of service restarts as they all try to handle the connection losses.  Windows' (mis?)handling of this scenario has been like this for so long that I doubt Microsoft would be willing to change it now.</li>
</ul>


<p>So I think that something, somewhere deep within Windows, is making Windows believe that it has lost the network layer.  Problem is, I don't (yet) know what's doing it - all I see is a lot of symptoms of it doing it, not the root cause.</p>

<p>"ipconfig /release && ipconfig /renew" will cause this (even if you immediately get the same IP address back), as will unplugging/replugging your Cat5 cable or disconnecting/reconnecting your WiFi connection, or power-saving on your NIC, or reconfiguring your NIC and thus forcing a reload of the driver, or...</p>

<p>We've yet to find the root cause in our setup, but investigations are ongoing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=331875&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-331875">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Still seems like a generic infrastructure issue symptom, which may be caused by Remoting defects in some cases.  Investigation from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pjdarton" class="user-hover" rel="pjdarton">pjdarton</a> is definitely correct, but I'd guess the most of reporters/voters here see quite another issue. Maybe it should be reported as a separate ticket since it may be worked around somehow.</p>

<p>I do not anticipate to have any time for Remoting maintenance anytime soon, so I am going to unassign this issue.</p>

</pre>
</details>

</details>
<p>Still seems like a generic infrastructure issue symptom, which may be caused by Remoting defects in some cases.  Investigation from <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pjdarton" class="user-hover" rel="pjdarton">pjdarton</a> is definitely correct, but I'd guess the most of reporters/voters here see quite another issue. Maybe it should be reported as a separate ticket since it may be worked around somehow.</p>

<p>I do not anticipate to have any time for Remoting maintenance anytime soon, so I am going to unassign this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pjdarton's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=331905&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-331905">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>FYI we got to the bottom of why our Windows slaves were disconnecting - it would appear that the Windows DHCP Client is incompatible with the Windows Time Service.</p>

<p>Our slaves were VMs created within OpenStack, and what we were seeing was a failure to renew the DHCP lease correctly.  When OpenStack detects that the guest OS has failed to renew the DHCP lease on time, it (briefly) drops the network link in order to prompt a lease renewal.  However this causes Windows to panic and kill all TCP connections (due to the way Windows mishandles network layers).
It seems that the DHCP client is not calculating the renewal time in a manner that's independent of the system's idea of "real time", and so it all goes wrong when the date/time gets changed (by the Windows Time Service), triggering OpenStack to bounce the physical layer, which Windows cascades in to an application-layer network outage killing the TCP connection that the slave relies on.</p>

<p>We "fixed" this by forcing our slaves to:</p>
<ol>
	<li>run "w32tm /resync" until they'd got the time synchronized,</li>
	<li>turn off the Windows Time Service entirely,</li>
	<li>ipconfig /release /renew to update the DHCP lease time</li>
	<li>start the Jenkins JNLP slave process</li>
</ol>


<p>This ensured that Windows would not update its clock while the slave's TCP connection was live, meaning that we weren't affected by the DHCP client's inability to keep the network alive after clock changes.
Since doing that we've not had any further problems of this nature (and we're quite pleased with that!)</p>

<p>Note: I've also seen Windows 10 report unpredictable (and incorrect) DHCP lease renewal times on other (non-OpenStack) machines - it lies.</p>
</pre>
</details>

</details>
<p>FYI we got to the bottom of why our Windows slaves were disconnecting - it would appear that the Windows DHCP Client is incompatible with the Windows Time Service.</p>

<p>Our slaves were VMs created within OpenStack, and what we were seeing was a failure to renew the DHCP lease correctly.  When OpenStack detects that the guest OS has failed to renew the DHCP lease on time, it (briefly) drops the network link in order to prompt a lease renewal.  However this causes Windows to panic and kill all TCP connections (due to the way Windows mishandles network layers).<br/>
It seems that the DHCP client is not calculating the renewal time in a manner that's independent of the system's idea of "real time", and so it all goes wrong when the date/time gets changed (by the Windows Time Service), triggering OpenStack to bounce the physical layer, which Windows cascades in to an application-layer network outage killing the TCP connection that the slave relies on.</p>

<p>We "fixed" this by forcing our slaves to:</p>
<ol>
	<li>run "w32tm /resync" until they'd got the time synchronized,</li>
	<li>turn off the Windows Time Service entirely,</li>
	<li>ipconfig /release /renew to update the DHCP lease time</li>
	<li>start the Jenkins JNLP slave process</li>
</ol>


<p>This ensured that Windows would not update its clock while the slave's TCP connection was live, meaning that we weren't affected by the DHCP client's inability to keep the network alive after clock changes.<br/>
Since doing that we've not had any further problems of this nature (and we're quite pleased with that!)</p>

<p>Note: I've also seen Windows 10 report unpredictable (and incorrect) DHCP lease renewal times on other (non-OpenStack) machines - it lies.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>franky4ro's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=331906&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-331906">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>After i did mass refactoring on my INFRA i do not see this type of errors anymore, except when the network in the lab goes high wire.</p>
</pre>
</details>

</details>
<p>After i did mass refactoring on my INFRA i do not see this type of errors anymore, except when the network in the lab goes high wire.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite's <a href="https://issues.jenkins.io/browse/JENKINS-28155?focusedId=432621&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432621">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Won't be fixed.  See <a href="https://issues.jenkins.io/browse/JENKINS-67604" title="WMI Windows Agent will no longer connect after Microsoft KB5004442 security update" class="issue-link" data-issue-key="JENKINS-67604"><del>JENKINS-67604</del></a> for the details of the deprecation of agents started by WMI calls using DCOM.</p>
</pre>
</details>

</details>
<p>Won't be fixed.  See <a href="https://issues.jenkins.io/browse/JENKINS-67604" title="WMI Windows Agent will no longer connect after Microsoft KB5004442 security update" class="issue-link" data-issue-key="JENKINS-67604"><del>JENKINS-67604</del></a> for the details of the deprecation of agents started by WMI calls using DCOM.</p>
